podcasts/Common SwiftUI/BannerView.swift

```diff
@@ -0,0 +1,113 @@
+    let iconName: String?
+    let invertedColor: Bool
+    let onActionTap: (() -> ())?
+    let onCloseTap: (() -> ())?
+
+    init(title: String? = nil, message: String? = nil, action: String? = nil, iconName: String? = nil, invertedColor: Bool = false, onActionTap: (() -> ())? = nil, onCloseTap: (() -> ())? = nil) {
+        self.title = title
+        self.message = message
+        self.action = action
+        self.iconName = iconName
+        self.invertedColor = invertedColor
+        self.onActionTap = onActionTap
+        self.onCloseTap = onCloseTap
+    }
+}
+
+struct BannerView: View {
+
+    @ObservedObject var model: BannerModel
+    @EnvironmentObject var theme: Theme
+
+    private var backgroundColor: Color {
+        if model.invertedColor {
+            if case .radioactive = theme.activeTheme {
+                return theme.primaryUi06
+            }
+            return theme.primaryUi02Active
+        }
+        switch theme.activeTheme {
+            case .indigo:
+                return theme.primaryUi02Active
+            case .contrastLight:
+                return theme.secondaryUi02
+            case .contrastDark:
+                return theme.primaryUi02Active
+            default:
+                return theme.primaryUi01
+        }
+    }
+
+    var body: some View {
+        HStack(alignment: .top) {
+            if let iconName = model.iconName {
+                Image(iconName)
+                    .foregroundColor(theme.primaryIcon03)
+            }
+            VStack(alignment: .leading, spacing: 8) {
+                if let title = model.title {
+                    Text(title)
+                        .font(.subheadline.weight(.medium))
+                        .foregroundColor(theme.primaryText01)
+                }
+                if let message = model.message {
+                    Text(message)
+                        .font(.caption2.weight(.semibold))
+                        .fixedSize(horizontal: false, vertical: true)
+                        .multilineTextAlignment(.leading)
+                        .foregroundColor(theme.primaryText02)
+                }
+                if let action = model.action {
+                    Button() {
+                        model.onActionTap?()
+                    } label: {
+                        Text(action)
+                            .font(.caption.weight(.semibold))
+                            .foregroundColor(theme.primaryInteractive01)
+                    }
+                }
+            }
+            Spacer()
+        }
+        .overlay(alignment: .topTrailing) {
+            if model.onCloseTap != nil {
+                Button() {
+                    model.onCloseTap?()
+                } label: {
+                    Image("close")
+                        .renderingMode(.template)
+                        .foregroundColor(theme.primaryIcon02)
+                }
+                .padding(8)
+            }
+        }
+        .padding()
```

Comment:
```suggestion
        .padding(.vertical, 20.0)
        .padding(.leading, 16.0)
        .padding(.trailing, model.onCloseTap != nil ? 56.0 : 16.0)
```
if the close button is present, we should probably add a trailing padding of 56 otherwise the `X` button will be too close to the text. The suggestion contains the value I used for the account creation

Reply:
Addressed here: https://github.com/Automattic/pocket-casts-ios/pull/2996/commits/6b2bf43c2fde4c76118e57bd08a0a281c80f7fae